src/FsToolkit.ErrorHandling/Seq.fs

```diff
@@ -10,47 +11,48 @@ module FsToolkit.ErrorHandling.Seq
 /// <param name="state">The initial state</param>
 /// <param name="f">The function to apply to each element</param>
 /// <param name="xs">The input sequence</param>
-/// <returns>A result with the ok elements in a sequence or the first error occurring in the sequence</returns>
+/// <returns>If state and all elements in 'sequence' are Ok, a result with the elements in an array. Alternately, the first error encountered</returns>
 let inline traverseResultM'
-    state
+    (state: Result<'okOutput seq, 'error>)
     ([<InlineIfLambda>] f: 'okInput -> Result<'okOutput, 'error>)
     (xs: 'okInput seq)
-    =
+    : Result<'okOutput[], 'error> =
+    if isNull xs then
+        nullArg (nameof xs)
+
     match state with
-    | Error _ -> state
-    | Ok oks ->
-        use enumerator = xs.GetEnumerator()
+    | Error e -> Error e
+    | Ok initialSuccesses ->
+
+        let oks = ResizeArray(initialSuccesses)
```

Comment:
NIT: I think we could go with an [ArrayCollector](https://fsharp.github.io/fsharp-core-docs/reference/fsharp-core-compilerservices-arraycollector-1.html) to benefit any small enough sequences. Might as well squeeze out all the tiniest perf benefits.

Reply:
Good point (my first time using it - it'd be nice to have a way to check for empty without the mutable flag abuse I added, though I suspect in terms of actual JITted IL this probably wins so not going to sweat it!

Reply:
That ArrayCollector shadowing behavior is probably something there should be a warnAsErrors setting or somesuch trapping?